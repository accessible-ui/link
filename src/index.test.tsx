/* jest */
import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import Link from './index'

describe('<Link>', () => {
  it('should render aria roles', () => {
    expect(
      render(
        <Link>
          <span onClick={() => (window.location.href = '/foo')} />
        </Link>
      ).asFragment()
    ).toMatchSnapshot(`role='link' tabindex='0'`)
  })

  it('should override default aria attributes', () => {
    expect(
      render(
        <Link>
          <span
            role="button"
            tabIndex={-1}
            onClick={() => (window.location.href = '/foo')}
          />
        </Link>
      ).asFragment()
    ).toMatchSnapshot(`role='button' tabindex='-1'`)
  })

  it('should work without onClick handler', () => {
    expect(
      render(
        <Link>
          <span role="button" tabIndex={-1} />
        </Link>
      ).asFragment()
    ).toMatchSnapshot(`renders`)
  })

  it('should call onClick when enter key is pressed', () => {
    const cb = jest.fn()
    const {getByTestId} = render(
      <Link>
        <span onClick={cb} data-testid="link" />
      </Link>
    )

    fireEvent.keyDown(getByTestId('link'), {which: 13})
    expect(cb).toBeCalledTimes(1)
  })
})
