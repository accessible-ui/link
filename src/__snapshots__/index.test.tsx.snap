// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> should override default aria attributes: role='button' tabindex='-1' 1`] = `
<DocumentFragment>
  <span
    role="button"
    tabindex="-1"
  />
</DocumentFragment>
`;

exports[`<Link> should render aria roles: role='link' tabindex='0' 1`] = `
<DocumentFragment>
  <span
    role="link"
    tabindex="0"
  />
</DocumentFragment>
`;

exports[`<Link> should work without onClick handler: renders 1`] = `
<DocumentFragment>
  <span
    role="button"
    tabindex="-1"
  />
</DocumentFragment>
`;
